Packs/FiltersAndTransformers/Scripts/TimeComponents/TimeComponents.py

```diff
@@ -0,0 +1,143 @@
+import locale
+import re
+from datetime import timezone, tzinfo
+import dateparser
+import demistomock as demisto  # noqa: F401
+import pytz
+from CommonServerPython import *  # noqa: F401
+
+
+def detect_time_zone(value: Any) -> tzinfo:
+    if isinstance(value, int):
+        return timezone(timedelta(minutes=value))
+
+    if isinstance(value, str):
+        value = value.strip()
+        if value == 'Z':
+            return timezone(timedelta(hours=0))
+        elif m := re.fullmatch('([+-])([0-9]{1,2})(:?([0-9]{2}))?', value):
+            # +H, +HMM, +HHMM, +H:MM, +HH:MM
+            offset = ((int(m[2]) * 60) + int(m[4] or 0)) * (-1 if m[1] == '-' else 1)
+            return timezone(timedelta(minutes=offset))
+        elif m := re.match(r'(^|\s)(UTC|GMT)([+-])([0-9]{1,2})(:?([0-9]{2}))?($|\s)', value):
+            # Includes: UTC+H, UTC+HMM, UTC+HHMM, UTC+H:MM, UTC+HH:MM
+            offset = ((int(m[4]) * 60) + int(m[6] or 0)) * (-1 if m[3] == '-' else 1)
+            return timezone(timedelta(minutes=offset))
+        else:
+            # Try to parse as zone info
+            try:
+                return pytz.timezone(value)
+            except Exception:
+                pass
+
+            # Try to parse as time string
+            try:
+                tz = dateparser.parse(value).tzinfo
+                if tz is not None:
+                    return tz
+            except Exception:
+                pass
+
+    raise DemistoException(f'Unable to extract time zone - {value}')
+
+
+def parse_date_time_value(value: Any) -> datetime:
+    """ Parse a date time value
+
+    :param value: The date or time to parse
+    :return: aware datetime object
+    """
+    if value is None or (isinstance(value, str) and not value):
+        return datetime.now(timezone.utc)
+
+    if isinstance(value, int):
+        # Parse as time stamp
+        try:
+            if value > 4294967295:
+                value /= 1000
+
+            date_time = datetime.fromtimestamp(value)
+            return date_time.astimezone(timezone.utc)
```

Comment:
parse_date_time_value() returns an aware datatime object, not a naive datetime object.

> :return: aware datetime object

It says to use datetime.fromtimestamp() instead of datetime.utcfromtimestamp().

https://docs.python.org/3/library/datetime.html

> classmethod datetime.utcfromtimestamp(timestamp)
> Return the UTC [datetime](https://docs.python.org/3/library/datetime.html#datetime.datetime) corresponding to the POSIX timestamp, with [tzinfo](https://docs.python.org/3/library/datetime.html#datetime.datetime.tzinfo) None. (The resulting object is naive.)
> 
> This may raise [OverflowError](https://docs.python.org/3/library/exceptions.html#OverflowError), if the timestamp is out of the range of values supported by the platform C gmtime() function, and [OSError](https://docs.python.org/3/library/exceptions.html#OSError) on gmtime() failure. It’s common for this to be restricted to years in 1970 through 2038.
> 
> To get an aware [datetime](https://docs.python.org/3/library/datetime.html#datetime.datetime) object, call [fromtimestamp()](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp):